server/src/main.rs

```diff
@@ -511,27 +507,57 @@ impl Database {
         &self,
         topic: &str,
         offset: u64,
-    ) -> Result<Vec<(u64, String, Vec<u8>)>, error::Error> {
-        let mut connection = self.connection.lock().await;
+    ) -> Result<
+        Option<(
+            u64,
+            String,
+            String,
+            Pin<Box<dyn Stream<Item = Result<Vec<u8>, error::Error>> + Send>>,
+        )>,
```

Comment:
This return type looks like it could cause confusion, it would be easy to swap the meaning of the strings. Would it be more explanatory to either create a type for the whole return, or for the meaning of the strings and u64?